pkg/dao/error.go

```diff
@@ -0,0 +1,11 @@
+package dao
+
+type Error struct {
+	Message       string
+	NotFound      bool
+	BadValidation bool
+}
+
+func (e *Error) Error() string {
+	return e.Message
+}
```

Comment:
this is a custom error struct that should allow us to pass more information up into the handler layer.  For example if we hit a uniquness constraint, we could tell the api that through this error struct.  that way it could return BadRequest rather than ISE 500